FTBFS on Linux i386 due to redefinition of af::dtype_traits<unsigned>

In an attempt to package the OpenCL backend for Debian, building the project for i386 raises the following error: 
```
In file included from /tmp/buildd/arrayfire-3.0.2/src/backend/opencl/Array.hpp:15:0,
                 from /tmp/buildd/arrayfire-3.0.2/src/backend/opencl/transpose.cpp:11:
/tmp/buildd/arrayfire-3.0.2/src/backend/opencl/traits.hpp: At global scope:
/tmp/buildd/arrayfire-3.0.2/src/backend/opencl/traits.hpp:36:8: error: redefinition of 'struct af::dtype_traits<unsigned int>'
 struct dtype_traits<size_t> {
        ^
In file included from /tmp/buildd/arrayfire-3.0.2/include/af/array.h:17:0,
                 from /tmp/buildd/arrayfire-3.0.2/src/backend/opencl/Array.hpp:12,
                 from /tmp/buildd/arrayfire-3.0.2/src/backend/opencl/transpose.cpp:11:
/tmp/buildd/arrayfire-3.0.2/include/af/traits.hpp:103:8: error: previous definition of 'struct af::dtype_traits<unsigned int>'
 struct dtype_traits<unsigned> {
        ^
src/backend/opencl/CMakeFiles/afopencl.dir/build.make:57: recipe for target 'src/backend/opencl/CMakeFiles/afopencl.dir/transpose.cpp.o' failed
```
Which likely comes from the fact that on i386, `size_t` is a typedef of `unsigned` for which there is already a definition inside `include/af/traits.hpp`.

The following fixed the issue, but seems quite fragile to me:
```
#if !defined(OS_WIN)        // Windows defines size_t as ulong
// temporary fix to avoid template redefinition on 32-bit architectures 
#if defined(__x86_64__) || defined(_M_X64) || defined(_WIN64) || defined(__aarch64__) || defined(__LP64__)
template<>
struct dtype_traits<size_t> {
    static const char* getName()
    {
        return (sizeof(size_t) == 8)  ? "ulong" : "uint";
    }
};
#endif
#endif
```

Besides, `size_t` is not OpenCL specific (unlike `clfloat` or `cldouble`) so it would make sense to move this definition out of the backend and place it in `include/af/traits.hpp`.

--- a/src/backend/opencl/traits.hpp
+++ b/src/backend/opencl/traits.hpp
@@ -31,16 +31,6 @@ struct dtype_traits<cl_double2> {
     static const char* getName() { return "double2"; }
 };
 
-#if !defined(OS_WIN)        // Windows defines size_t as ulong
-template<>
-struct dtype_traits<size_t> {
-    static const char* getName()
-    {
-        return (sizeof(size_t) == 8)  ? "ulong" : "uint";
-    }
-};
-#endif
-
 template<typename T> static bool iscplx() { return false; }
 template<> STATIC_ bool iscplx<cl_float2>() { return true; }
 template<> STATIC_ bool iscplx<cl_double2>() { return true; }

